pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -193,14 +194,56 @@ func (r *OperatorStepResolver) ResolveSteps(namespace string) ([]*v1alpha1.Step,
 			})
 		}
 
-		// add steps for subscriptions for bundles that were added through resolution
+		// add steps for subscriptions for bundles that were added through resolution and bundles that have been marked deprecated
 		for sub := range existingSubscriptions {
-			if sub.Status.CurrentCSV == op.Name {
-				continue
+			updated := false
+			if len(op.Deprecations) == 0 && sub.Status.GetCondition(v1alpha1.SubscriptionDeprecated).Status != corev1.ConditionUnknown {
+				// If new bundle is no longer deprecated then remove the conditions
+				sub.Status.RemoveConditions(v1alpha1.SubscriptionDeprecated, v1alpha1.SubscriptionPackageDeprecated, v1alpha1.SubscriptionChannelDeprecated, v1alpha1.SubscriptionBundleDeprecated)
+				updated = true
+			} else if len(op.Deprecations) > 0 {
+				// Remove old conditions first in case of update
+				sub.Status.RemoveConditions(v1alpha1.SubscriptionDeprecated, v1alpha1.SubscriptionPackageDeprecated, v1alpha1.SubscriptionChannelDeprecated, v1alpha1.SubscriptionBundleDeprecated)
+				rollupMessage := ""
+				for _, deprecation := range op.Deprecations {
+					switch deprecation.Type {
+					case declcfg.SchemaPackage:
+						sub.Status.SetCondition(v1alpha1.SubscriptionCondition{
+							Type:    v1alpha1.SubscriptionPackageDeprecated,
+							Message: deprecation.Message,
+							Status:  corev1.ConditionTrue,
+						})
+					case declcfg.SchemaChannel:
+						sub.Status.SetCondition(v1alpha1.SubscriptionCondition{
+							Type:    v1alpha1.SubscriptionChannelDeprecated,
+							Message: deprecation.Message,
+							Status:  corev1.ConditionTrue,
+						})
```

Comment:
It looks like `subscription.spec.channel` is an optional field. We should only set the channel deprecation condition if `subscription.spec.channel` is set and that channel is deprecated.